frontend/src/components/DataProductTable.vue

```diff
@@ -0,0 +1,184 @@
+              <b-spinner class="align-middle"></b-spinner>
+              <strong>Loading...</strong>
+            </div>
+          </template>
+
+        </b-table>
+      </div>
+    </div>
+
+    <div class="text-right">
+      <b-row class="mb-4">
+        <b-col sm="7" md="3" class="my-1 ml-auto pr-5">
+          <b-pagination
+              v-model="currentPage"
+              :total-rows="totalRows"
+              :per-page="perPage"
+              prev-text="< Prev"
+              next-text="Next >"
+              align="right"
+              size="sm"
+              pills
+              class="my-0"
+          ></b-pagination>
+        </b-col>
+      </b-row>
+    </div>
+  </b-container>
+</template>
+
+<script>
+import {smilesDPService} from "@/services/smiles-dp-service";
+import {configurationService} from "@/services/configuraion-service";
+
+export default {
+  name: "SMILESDataProductTable",
+  data() {
+    return {
+      items: [],
+      totalRows: 1,
+      currentPage: 1,
+      perPage: 15,
+      filter: null,
+      type: this.$route.query.type,
+      isBusy: true,
+    }
+  },
+  computed: {
+    fields() {
+      return configurationService.getDisplayableColumns(this.type);
+    },
+    formattedJson() {
+      return (row) => {
+        return JSON.stringify(row.item, null, 2);
+      };
+    }
+  },
+  mounted() {
+    this.loadSMILESDataProducts();
+  },
+  methods: {
+    loadSMILESDataProducts() {
+      this.params = `page=${this.currentPage}&size=${this.perPage}`;
+      smilesDPService.getSMILESDataProducts(this.type, this.params).then((response) => {
+        if (response.data) {
+          this.items = response.data;
+          // Set the initial number of items
+          this.totalRows = this.items.length
+        }
+      });
+      this.isBusy = false;
+    },
+    onFiltered(filteredItems) {
+      // Trigger pagination to update the number of buttons/pages due to filtering
+      this.totalRows = filteredItems.length
+      this.currentPage = 1
+    },
+    deleteRecord(data) {
+      this.$bvModal
+          .msgBoxConfirm("Do you want to delete the record?", {
+            title: "Record Delete Confirmation",
+            size: "mm",
+            buttonSize: "sm",
+            okVariant: "danger",
+            okTitle: "YES",
+            cancelTitle: "NO",
+            footerClass: "p-2",
+            hideHeaderClose: false,
+            centered: true,
+          })
+          .then((value) => {
+            console.log(data.item.data_product_id)
+            if (value) {
+              smilesDPService.deleteSMILESDataProduct(this.type, data.item.data_product_id)
+              const index = this.items.findIndex(item => item.data_product_id === data.item.data_product_id);
+              if (index !== -1) {
+                this.items.splice(index, 1);
+              }
+            }
+          });
+    },
+    edit(data) {
+      alert(JSON.stringify(data));
```

Comment:
Not sure what edit should do here, but let's not use `alert()`. Use a modal or some other BootstrapVue component if you need a popup window.

Reply:
The edit functionality will be removed from this Vue component and instead, it will be added to the detailed page